pkg/cmd/ssh/options.go

```diff
@@ -151,6 +151,104 @@ var (
 
 		return cmd.Run()
 	}
+
+	// waitForSignal informs the user about their SSHOptions and keeps the
+	// bastion alive until gardenctl exits.
+	waitForSignal = func(ctx context.Context, o *SSHOptions, shootClient client.Client, bastion *operationsv1alpha1.Bastion, nodeHostname string, nodePrivateKeyFiles []string, signalChan <-chan struct{}) error {
```

Comment:
As far as I understand the code this function makes no assumption that the role of the cluster is a shoot cluster. I would rename the parameter from `shootClient` to `client` and remove the word shoot in all messages. What do you think?